crates/egui/src/context.rs

```diff
@@ -1066,28 +1065,31 @@ impl Context {
                                     interaction.drag_id = Some(id);
                                     interaction.drag_is_window = false;
                                     memory.set_window_interaction(None); // HACK: stop moving windows (if any)
-
-                                    res.is_pointer_button_down_on = true;
+                                    res
+                                        .state
+                                        .insert(ResponseState::IS_POINTER_BUTTON_DOWN_ON);
 
                                     // Again, only if we are ONLY sensitive to drags can we decide that this is a drag now.
                                     if sense.click {
-                                        res.dragged = false;
-                                        res.drag_started = false;
+                                        res.state.remove(ResponseState::DRAGGED);
+                                        res.state.remove(ResponseState::DRAG_STARTED);
                                     } else {
-                                        res.dragged = true;
-                                        res.drag_started = true;
+                                        res.state.insert(ResponseState::DRAGGED);
+                                        res.state.insert(ResponseState::DRAG_STARTED);
                                     }
                                 }
                             }
                         }
 
                         PointerEvent::Released { click, button } => {
-                            res.drag_released = res.dragged;
-                            res.dragged = false;
+                            res
+                                .state
+                                .set(ResponseState::DRAG_RELEASED, res.dragged());
+                            res.state.remove(ResponseState::DRAGGED);
 
-                            if sense.click && res.hovered && res.is_pointer_button_down_on {
+                            if sense.click && res.hovered() && res.is_pointer_button_down_on() {
                                 if let Some(click) = click {
-                                    let clicked = res.hovered && res.is_pointer_button_down_on;
+                                    let clicked = res.hovered() && res.is_pointer_button_down_on();
```

Comment:
Since clicked now has more complexity associated with it (performing bitwise operations to check if the bitflags contain those 2), should it be moved out of the 2 ifs and just reused? Or is the added cycles negligible to performance?

Reply:
The optimizer should take care of it for you if `fn hovered` is marked `#[inline]`